src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param MetadataInterface $metadata
+     */
+    protected function setClassesParameters(ContainerBuilder $container, MetadataInterface $metadata)
+    {
+        if ($metadata->hasClass('model')) {
+            $container->setParameter(sprintf('%s.model.%s.class', $metadata->getApplicationName(), $metadata->getName()), $metadata->getClass('model'));
+        }
+        if ($metadata->hasClass('controller')) {
+            $container->setParameter(sprintf('%s.controller.%s.class', $metadata->getApplicationName(), $metadata->getName()), $metadata->getClass('controller'));
+        }
+        if ($metadata->hasClass('factory')) {
+            $container->setParameter(sprintf('%s.factory.%s.class', $metadata->getApplicationName(), $metadata->getName()), $metadata->getClass('factory'));
+        }
+        if ($metadata->hasClass('repository')) {
+            $container->setParameter(sprintf('%s.repository.%s.class', $metadata->getApplicationName(), $metadata->getName()), $metadata->getClass('repository'));
+        }
+
+        if (!$metadata->hasParameter('validation_groups')) {
+            return;
+        }
+
+        $validationGroups = $metadata->getParameter('validation_groups');
+
+        foreach ($validationGroups as $formName => $groups) {
+            $suffix = ('default' === $formName ? '' : sprintf('_%s', $formName));
+            $container->setParameter(sprintf('%s.validation_groups.%s%s', $metadata->getApplicationName(), $metadata->getName(), $suffix), $groups);
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param MetadataInterface $metadata
+     */
+    protected function addController(ContainerBuilder $container, MetadataInterface $metadata)
+    {
+        // @todo: Remove when ResourceController is reworked.
+        $configurationDefinition = new Definition(new Parameter('sylius.controller.configuration.class'));
+        $configurationDefinition
+            ->setFactory(array(new Reference('sylius.controller.configuration_factory'), 'createConfiguration'))
+            ->setArguments(array($metadata->getApplicationName(), $metadata->getName(), $metadata->getTemplatesNamespace()))
+            ->setPublic(false)
+        ;
+
+        $definition = new Definition($metadata->getClass('controller'));
+        $definition
+            ->setArguments(array($configurationDefinition))
+            ->addMethodCall('setContainer', array(new Reference('service_container')))
+        ;
+
+        $container->setDefinition($metadata->getServiceId('controller'), $definition);
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param MetadataInterface $metadata
+     */
+    protected function addFactory(ContainerBuilder $container, MetadataInterface $metadata)
+    {
+        $translatableFactoryInterface = 'Sylius\Component\Translation\Factory\TranslatableFactoryInterface';
```

Comment:
Translation component is optional at this point, I will open an RFC about including it in the Resource, but for now we can't assume that the interface exists.